### ISSUE TITLE
[Feature Request] Support for CLI, or a plugin/script API

### ISSUE THREAD

**Issue by buanzo** [zencq/NomNom#101]
It would be very convenient to have a way to script/automate some operations.

**Examples:**
```
nomnom.exe saves copy --source=4 --dest=7
nomnom.exe backup --source=4 --format=json --dest=savegame.json
nomnom.exe restore --format=json --from=savegame.json --dest=4
```
Or maybe implement LUA, and be able to run scripts:

nomnom.exe script run somescript.nom

Thank you!

**Comment by cengelha** [comment]
There is already a simple CLI to convert saves, but it may not work in all cases tho. Therefore, I'll keep this open for now as a reminder to add more options in the future.

Current options:
```
"input", "in", "i" => input file
"output", "out", "o" => output file
"json", "j" => convert to json
  "indent", "d" => indent the result
  "mapped", "map", "p" => map/deobfuscate the result
"save", "s" => convert to save
  "platform", "f" => target platform
    "gog.com", "gogcom", "gog"
    "microsoft", "ms", "xbox", "xb", "x"
    "sony", "playstation", "ps", "ps4"
    "steam", "st"
    "nintendo", "switch", "sw"
```

**Comment by cengelha** [comment]
@buanzo It's been a while. I decided to add a CLI to the underlying library of mine instead of NN: [libNOM.io](https://github.com/zencq/libNOM.io)
It will be available from v0.8.0 on. Might not be perfect at the beginning but I'll improve it in the future.
Closing this one now.
